src/System.Management.Automation/engine/AutomationEngine.cs

```diff
@@ -35,6 +35,34 @@ internal class AutomationEngine
         /// </summary>
         internal AutomationEngine(PSHost hostInterface, InitialSessionState iss)
         {
+#if !UNIX
+            // Update the env variable PathEXT to contain .CPL
+            var pathext = Environment.GetEnvironmentVariable("PathEXT");
+            pathext = pathext ?? string.Empty;
+            bool cplExist = false;
+            if (pathext != string.Empty)
+            {
+                string[] entries = pathext.Split(Utils.Separators.Semicolon);
+                foreach (string entry in entries)
+                {
+                    string ext = entry.Trim();
+                    if (ext.Equals(".CPL", StringComparison.OrdinalIgnoreCase))
```

Comment:
Can we simplify the code with `cplExist  = pathext.Contains(".CPL", StringComparison.OrdinalIgnoreCase)` ?

Reply:
Technically, it's possible there is an extension `.CPL2` that would cause such a simple search to fail.

Reply:
Good catch!

I looked more and now I wonder why do we change the environment variable at all? An user can change this variable himself if he wants. If PowerShell need this it can do this internally only.
Can we add new extension here like ".ps1":
https://github.com/PowerShell/PowerShell/blob/bd6fdae73520931f0d27a29d6290e18761772141/src/System.Management.Automation/engine/CommandDiscovery.cs#L1419-L1432

Reply:
I don't know the reason it was done previously, but it does break a usage scenario from Windows PowerShell if we don't retain this.  As for `.ps1`, it's special cased by PowerShell so it doesn't need to be in `PATHEXT`.  If it wasn't special cased by PowerShell and handled by `PATHEXT`, I would expect it to start a new PowerShell process rather than executing in the current one.

Reply:
> but it does break a usage scenario from Windows PowerShell if we don't retain this

What scenario you mean? This can only affect a sub-process, but which application launched from PowerShell needs to have .CPL in EXTPATH? It seems EXTPATH is for PowerShell itself only and we can safely add .CPL directly in ext cache.

Reply:
@iSazonov the scenario is starting `*.cpl` control panels without typing the extension like `firewall`.

Reply:
The scenario will work if we add the extension in cache without modifying environment variable, yes?

Reply:
We could special case cpl like we special case ps1 although the use case is not the same (the first is still out-of-proc while the latter is in-proc).  `PATHEXT` is exactly for the cpl use case, so I don't see a reason to do this in code when Windows already handles this for us if we add the extension to the env var.

Reply:
@SteveL-MSFT Sorry I am so picky. It is my fresh look on the code and I very wonder how it works. Also it is on _hot path_ for searching external commands.

> the first is still out-of-proc while the latter is in-proc

Out-of-proc makes no sense for PowerShell.
PATHEXT does not contain ".CPL" by default on Windows
 - appwiz.cpl run from cmd.exe and PowerShell without the fix
 - _appwiz doesn't run from cmd.exe and PowerShell without the fix_
So only in-proc works.

A scenario we would break is like - start cmd.exe in PowerShell session and run "appwiz".
I am sure that this was not the purpose of this design but only random side effect - 
if it would be necessary for users in cmd.exe it was on the system by default, or should be added by the users themselves.
For non-interactive scenarios the lack of extensions is secure dangerous and we should not encourage it at all.

Reply:
@iSazonov my primary concern is that we are positioning PS7 as a viable replacement for WinPS 5.1 and there has already been one customer complaint that this is no longer working (despite it being different from cmd.exe, but WinPS is default shell on Windows 10).

Reply:
It seems we say about different things.
> PATHEXT is exactly for the cpl use case, so I don't see a reason to do this in code when Windows already handles this for us if we add the extension to the env var.

It is _wrong_. As I show above the out-of-proc doesn't work in the scenario. _We do all work internally_. Modifing PATHEXT is an extra operation which is not associated with this scenario.
Sorry again for the PR delay. I wouldn’t want us to pick up delays again when starting PowerShell/runspaces.